pkg/edit/repl.go

```diff
@@ -0,0 +1,51 @@
+package edit
+
+// This file encapsulates functionality related to a complete REPL cycle. Such as capturing
+// information about the most recently executed interactive command.
+
+import (
+	"src.elv.sh/pkg/eval"
+	"src.elv.sh/pkg/eval/vals"
+	"src.elv.sh/pkg/eval/vars"
+)
+
+//elvdoc:var after-command
+//
+// A list of functions to call after each interactive command completes. There is one pre-defined
+// function used to populate the [`$edit:command-duration`](./edit.html#editcommand-duration)
+// variable. Each function is called with a single pseudo-map argument containing the following
```

Comment:
> a single pseudo-map argument

The implementation uses a real map

Reply:
Good catch. Originally I intended to use a struct map but ultimately decided a real map was simpler and potentially useful to allow the hook to modify the content.

Reply:
I just realized making this a real map isn't particularly useful given the [semantics of mutating maps](https://elv.sh/ref/language.html#set) and [assigning to containers](https://elv.sh/learn/unique-semantics.html#assignment-semantics). I was thinking that making it a real map would allow an earlier hook to affect the data seen by a later hook. It's arguably a good thing that isn't possible from a security perspective. I'm going to leave it a real map for the moment. It is easy enough to make it a lighter weight struct map.